AdobeStockImage/Test/Integration/Model/Extract/DocumentToAssetTest.php

```diff
@@ -0,0 +1,125 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\AdobeStockImage\Test\Integration\Model\Extract;
+
+use Magento\AdobeStockAssetApi\Api\Data\AssetInterface;
+use Magento\AdobeStockAssetApi\Api\Data\CategoryInterface;
+use Magento\AdobeStockAssetApi\Api\Data\CreatorInterface;
+use Magento\AdobeStockImage\Model\Extract\AdobeStockAsset;
+use Magento\Framework\Api\AttributeInterface;
+use Magento\Framework\Api\Search\Document;
+use Magento\TestFramework\Helper\Bootstrap;
+use PHPUnit\Framework\TestCase;
+
+/**
+ * Test converting a Document from the search result to Adobe Stock asset
+ */
+class DocumentToAssetTest extends TestCase
+{
+    /**
+     * @var AdobeStockAsset
+     */
+    private $documentToAsset;
+
+    /**
+     * Prepare test objects.
+     */
+    protected function setUp(): void
+    {
+        $this->documentToAsset = Bootstrap::getObjectManager()->get(
+            AdobeStockAsset::class
+        );
+    }
+
+    /**
+     * @dataProvider documentProvider
+     * @param array $assetData
+     * @param array $categoryData
+     * @param array $creatorData
+     * @param Document $document
+     * @param array $additionalData
+     */
+    public function testConvert(
+        array $assetData,
+        array $categoryData,
+        array $creatorData,
+        Document $document,
+        array $additionalData
+    ): void {
+        $this->documentToAsset = Bootstrap::getObjectManager()->get(
+            AdobeStockAsset::class
+        );
+        $documentToAsset = $this->documentToAsset->convert($document, $additionalData);
```

Comment:
Naming clarification suggestion
```suggestion
        $asset = $this->documentToAsset->convert($document, $additionalData);
```